aidial_adapter_bedrock/llm/model/stability/v2.py

```diff
@@ -27,10 +30,57 @@
 from aidial_adapter_bedrock.llm.model.stability.storage import save_to_storage
 from aidial_adapter_bedrock.llm.truncate_prompt import DiscardedMessages
 from aidial_adapter_bedrock.utils.json import remove_nones
+from aidial_adapter_bedrock.utils.resource import Resource
 
 SUPPORTED_IMAGE_TYPES = ["image/jpeg", "image/png", "image/webp"]
 
 
+def _validate_image_size(
+    image: Resource,
+    width_constraints: Tuple[int, int] | None,
+    height_constraints: Tuple[int, int] | None,
+) -> None:
+    if width_constraints is None and height_constraints is None:
+        return
+
+    with Image.open(BytesIO(image.data)) as img:
+        width, height = img.size
+        if width_constraints and not (
```

Comment:
Would have been great to reduce code duplication:

```python
for rng, val, name in [(width_constraints, width, "width"), (height_constraints, height, "height")]:
    if rng is None:
        continue
    (min_val, max_val) = val
    ...
```

Reply:
Good idea,done